src/com/intel/cancelPreviousBuildsInternal.groovy

```diff
@@ -0,0 +1,35 @@
+// src/com/intel/cancelPreviousBuildsInternal.groovy
+
+package com.intel
+
+/**
+ * cancelPreviousBuildsInternal.groovy
+ *
+ * Routine to canel old builds in progress
+ */
+
+
+def cancelPreviousBuildsInternal(Map config = [:]) {
+  /**
+   * Cancel old builds method.
+   *
+   * @param config Map of parameters passed (currently none)
+   * @return Nothing
+   */
+
+  def jobName = env.JOB_NAME
+  def buildNumber = env.BUILD_NUMBER.toInteger()
+  /* Get job name */
+  def currentJob = Jenkins.instance.getItemByFullName(jobName)
+
+  /* Iterating over the builds for specific job */
+  for (def build : currentJob.builds) {
+    /* If there is a build that is currently running and it's not current build */
+    if (build.isBuilding() && build.number.toInteger() != buildNumber) {
```

Comment:
I feel sure there should be a < rather than a != in here somewhere.

If the intention is that this should be called early on in the pipeline then that should be added as a comment.

Reply:
I'm not sure there is anything requiring that it be called early in the pipeline.  For the use-case I (seems we) have in mind, it would be done early and that's the use-case I tested.

But I think if somebody had a use-case where it only happen after passing certain stages, say, Build for example, then I don't think there would be any harm in doing it there instead.